kernel/src/main/scala/cats/kernel/Order.scala

```diff
@@ -220,15 +218,73 @@ object Order extends OrderFunctions[Order] with OrderToOrderingConversion {
     }
 
   def fromOrdering[A](implicit ev: Ordering[A]): Order[A] =
-    new Order[A] {
-      def compare(x: A, y: A): Int = ev.compare(x, y)
-
-      override def toOrdering: Ordering[A] = ev
-    }
+    OrderFromOrdering[A](ev)
 
   def fromComparable[A <: Comparable[A]]: Order[A] =
     new Order[A] {
       override def compare(x: A, y: A): Int =
         x.compareTo(y)
     }
 }
+
+/** Wrapper class used to track when we create an `Order` instance from an
+  * `Ordering` instance. This allows us to know that given `Order` instance is
+  * consistent with the given `Ordering` instance. This becomes important when
+  * dealing with structures which rely on `Ordering`, e.g. `SortedSet`.
+  */
+final private[kernel] class OrderFromOrdering[A] private (val value: Ordering[A]) extends Order[A] {
```

Comment:
any reason not to just make it a case class and get the equals/hashCode for free?

Reply:
We can do that. I usually just think case classes for data, classes for typeclasses and F-Algebras, but in this case we are blurring those lines.